addons/delivery/models/delivery_carrier.py

```diff
@@ -123,3 +146,114 @@ def cancel_shipment(self, pickings):
             return getattr(self, '%s_cancel_shipment' % self.delivery_type)(pickings)
+
+    @api.onchange('state_ids')
+    def onchange_states(self):
+        self.country_ids = [(6, 0, self.country_ids.ids + self.state_ids.mapped('country_id.id'))]
+
+    @api.onchange('country_ids')
+    def onchange_countries(self):
+        self.state_ids = [(6, 0, self.state_ids.filtered(lambda state: state.id in self.country_ids.mapped('state_ids').ids).ids)]
+
+    @api.multi
+    def verify_carrier(self, contact):
+        self.ensure_one()
+        if self.country_ids and contact.country_id not in self.country_ids:
+            return False
+        if self.state_ids and contact.state_id not in self.state_ids:
+            return False
+        if self.zip_from and (contact.zip or '') < self.zip_from:
+            return False
+        if self.zip_to and (contact.zip or '') > self.zip_to:
+            return False
+        return self
+
+    @api.multi
+    def create_price_rules(self):
+        PriceRule = self.env['delivery.price.rule']
+        for record in self:
+            # If using advanced pricing per destination: do not change
+            if record.delivery_type == 'base_on_rule':
+                continue
+
+            # Not using advanced pricing per destination: override lines
+            if record.delivery_type == 'base_on_rule' and not (record.fixed_price is not False or record.free_if_more_than):
+                record.price_rule_ids.unlink()
+
+            # Check that float, else 0.0 is False
+            if not (record.fixed_price is not False or record.free_if_more_than):
+                continue
+
+            if record.delivery_type == 'fixed':
+                PriceRule.search([('carrier_id', '=', record.id)]).unlink()
+
+            line_data = {
+                'carrier_id': record.id,
+                'variable': 'price',
+                'operator': '>=',
+            }
+            # Create the delivery price rules
+            if record.free_if_more_than:
+                line_data.update({
+                    'max_value': record.amount,
+                    'standard_price': 0.0,
+                    'list_base_price': 0.0,
+                })
+                PriceRule.create(line_data)
+            if record.fixed_price is not False:
+                line_data.update({
+                    'max_value': 0.0,
+                    'standard_price': record.fixed_price,
+                    'list_base_price': record.fixed_price,
+                })
+                PriceRule.create(line_data)
+        return True
+
+    @api.model
+    def create(self, vals):
+        res = super(DeliveryCarrier, self).create(vals)
+        res.create_price_rules()
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super(DeliveryCarrier, self).write(vals)
+        self.create_price_rules()
+        return res
+
+    @api.multi
+    def get_price_available(self, order):
```

Comment:
I don't understand the name of this function. What is the meaning of "available" in "get_price_available" ?